File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = []
    i, n = 0, len(s)

    while i < n:
        if answer and answer[-1] == s[i]:
            answer.pop()
        else:
            answer.append(s[i])
        i += 1

    return 1 if not answer else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;


class Solution
{
    public int solution(String s) {
        final Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            if (!stack.isEmpty() && stack.peek() == s.charAt(i)) {
                stack.pop();
            } else {
                stack.push(s.charAt(i));
            }
        }

        int answer = 0;

        if (stack.isEmpty()) {
            answer = 1;
        }
        return answer;
    }
}